front_end/src/backend/uploads.js

```diff
@@ -0,0 +1,151 @@
+/**
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import axios from 'axios';
+import { REGIONS_MAP, FILESIZE_BYTES, FILESIZE_MIB, FLOAT_ROUND_DIGITS, DEFAULT_TIME_TAKEN } from './common';
+
+/**
+ * Strings of fixed length to be used as contents of an 
+ * uploaded file during an upload benchmark.
+ * @enum {string}
+ */
+const FILE_CONTENTS = Object.freeze({
+    '2mib.txt': '\x00'.repeat(2097152),
+    '64mib.txt': '\x00'.repeat(67108864),
+    '256mib.txt': '\x00'.repeat(268435456),
+});
+
+/**
+ * MD5 Checksums of contents of each string that is uploaded.
+ * @enum {string}
+ */
+const MD5_SUMS = Object.freeze({
+    '2mib.txt': 'stEjbChqPAcEIk/kEF7KSQ==',
+    '64mib.txt': 'f2FNqTKc066/WbkarcML8A==',
+    '256mib.txt': 'H1A55QvWaykMVmhNhVDGwg=='
+});
+
+/**
+ * Measure time to upload files to a bucket from memory and get relevant benchmarks.
+ */
+export class Uploads {
+    /**
+     * Get signed URL to be used to upload a file through a PUT request. 
+     * 
+     * @param {string} fileName 
+     * @param {string} bucketName 
+     * @returns {string} Returns a signed URL.
+     */
+    async getSignedURL(fileName, bucketName) {
+        // The signed URL Source is set to localhost now, but will either use env variables or another website during deployment
+        const signedUrlSource = `http://localhost:3000/${bucketName}/${fileName}`;
+
+        const axiosOptions = {
+            method: 'GET',
+            url: signedUrlSource,
+        }
+
+        const response = await axios(axiosOptions);
```

Comment:
Not sure if it is handled at a level above this or not but this can potentially throw an exception. If it is not handled at a layer above you should wrap this in try/catch

Reply:
Yeap, this is handled a layer above